### PR TITLE
Improve slow and unresponsive query troubleshooting docs & refactor old troubleshooting docs

### PR DESCRIPTION
**Pull request by chaas**
Based on the figjam diagram linked in https://github.com/MaterializeInc/materialize/issues/23322.
Create separate troubleshooting pages for:
* Slow & unresponsive queries
* Ingest
* Sinks
* Dataflows

For all but " Slow & unresponsive queries", I just seeded the page with existing content from the old huge `/manage/troubleshooting.md` file. Each owning team should do additional follow up work to improve those pages.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: doc/user/content/transform-data/source-troubleshooting.md
Comment (chaas) [reply doc/user/content/transform-data/source-troubleshooting.md:8]: Ah I wonder if the slow/unresponsive queries should go under a troubleshooting doc in serve-results then, rather than transform-data. And the file here "dataflow-troubleshooting.md" stays in transform-data.

**Comment by chaas** [comment]
@morsapaes This is ready for another round of review. It should be more close to merge-ready at this point!

I got rid of the original /manage/troublehshooting.md file and fanned out the contents into: slow query, ingest, dataflow, and sink -specific troubleshooting pages.

File: doc/user/content/transform-data/source-troubleshooting.md
Comment (morsapaes) [reply doc/user/content/transform-data/source-troubleshooting.md]: That's...a great question. The fact that I don't like the current navigation (🫠 ) makes it harder to answer, but it makes sense to place it under `Serve results` for now, yeah!

**Comment by morsapaes** [comment]
This is great — thank you so much for working on it, @chaas! Regarding redirects, we need to keep these aliases around:

```
aliases:
  - /ops/diagnosing-using-sql/
  - /ops/troubleshooting/
```

If you're done with the bulk of the work, I can take over and take care of the redirects, moving stuff around, and making things consistent with the rest of the documentation so we can merge this soon.

**Comment by chaas** [comment]
> If you're done with the bulk of the work, I can take over and take care of the redirects, moving stuff around, and making things consistent with the rest of the documentation so we can merge this soon.

Yeah I think this is good enough for now, and I can make incremental improvements once we get the base merged. Let me know if I can help with anything in getting this merged.

Also, do we want a review by field eng that this guidance is in line with how they want to direct customers?

File: doc/user/content/ingest-data/ingest-troubleshooting.md
Comment (sjwiesman) [inline doc/user/content/ingest-data/ingest-troubleshooting.md:36]: Let's add a sentence defining what the initial snapshot is and why it has to be committed atomically.

File: doc/user/content/transform-data/troubleshooting.md
Comment (sjwiesman) [inline doc/user/content/transform-data/troubleshooting.md:38]:
```suggestion
*Do you have multiple materialized views chained on top of each other? Are you seeing small amounts of lag?*
```
Extra `\`?

File: doc/user/content/transform-data/troubleshooting.md
Comment (sjwiesman) [inline doc/user/content/transform-data/troubleshooting.md:39]: Shout this be in a `note` shortcode?

**Comment by chuck-alt-delete** [comment]
Thank you for this PR! There’s a lot more valuable troubleshooting steps now.

Here are some of my [go-to queries](https://www.notion.so/materialize/Observability-Queries-ea2783099b6e42a18c169b154c1f88ba?pvs=4) for troubleshooting environments if those are helpful. A lot of these now have console UI equivalents, but it can be helpful to see both UI and query side by side to teach users how to use system catalog.

File: doc/user/content/transform-data/troubleshooting.md
Comment (chaas) [reply doc/user/content/transform-data/troubleshooting.md:38]: That's markdown syntax for a newline without a new paragraph.
Without that it would wrap the "Tip:..." text into the previous line.

@morsapaes is there a more proper way to do that in markdown?

File: doc/user/content/ingest-data/ingest-troubleshooting.md
Comment (chaas) [reply doc/user/content/ingest-data/ingest-troubleshooting.md:36]: Good suggestion. I added a small explainer.
I took content from https://materialize.com/docs/ingest-data/postgres-alloydb/, which is a great walkthru. @morsapaes maybe we can figure out how to fold this walkthru into the troubleshooting docs somehow and/or make it more generic?

**Comment by chaas** [comment]
> my [go-to queries](https://www.notion.so/materialize/Observability-Queries-ea2783099b6e42a18c169b154c1f88ba?pvs=4) for troubleshooting environments

Thanks @chuck-alt-delete ! I erred on the side of only pointing to the console if the info is available there, but I'd be interested to hear how often users want to run the SQL queries vs prefer looking at the console based on your experience, in particular after we release these docs. Want to understand how the average user interacts with Materialize.

File: doc/user/content/ingest-data/ingest-troubleshooting.md
Comment (morsapaes) [reply doc/user/content/ingest-data/ingest-troubleshooting.md:36]: I'm hopeful that we can soon just point users to the console. This will be exposed as part of the source progress work, right?

File: doc/user/content/transform-data/troubleshooting.md
Comment (morsapaes) [reply doc/user/content/transform-data/troubleshooting.md:38]: I typically use `<br/>`, but `\` works too!

File: doc/user/content/ingest-data/ingest-troubleshooting.md
Comment (chaas) [reply doc/user/content/ingest-data/ingest-troubleshooting.md:36]: Ah yeah I think so. We're also already in private preview for query history in the console, so will add content for both of those soon. I just added a phase 2 to the troubleshooting docs tracking ticket for all the follow up items like query history in console, source progress in console, and incorporating all of chuck's queries.

Comment (morsapaes) [reply doc/user/content/ingest-data/ingest-troubleshooting.md:36]: Oh, now that I'm going through it, I understand what you mean by "making it more generic". Took a stab at it:

> When you create a source, Materialize takes a snapshot of any existing data in
the upstream external system and ingests that snapshotted data before it starts
ingesting new data. To ensure correct and consistent results, this snapshot is
committed atomically at a specific timestamp. Because of that, you will not be
able to query your source (or, queries will return no data) until Materialize
has finished ingesting the initial snapshot.

Comment (morsapaes) [reply doc/user/content/ingest-data/ingest-troubleshooting.md:36]: You don't need to work this in, @chaas! I started reviewing this and will just push a commit with this sort of edits. I'll ping you for one last look tomorrow, and 🤞 we're all set to merge then!

**Comment by morsapaes** [comment]
@chaas: moved the dataflow troubleshooting guide to a standalone page that links from the `Transform data` guide. There's some questions in there that can be transferred to the main guide, but it overall feels like a level too deep for users as is.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**
